### PR TITLE
Support DNS certificate provision and management

### PR DESCRIPTION
**Pull request by lei-tang**
This PR is for issue: https://github.com/istio/istio/issues/17061
This PR depends on: https://github.com/istio/istio/pull/17400

Design documents:
[1]https://docs.google.com/document/d/1VHyBre8943USOx_YEVtA18KfEoGTmLT1iBHESO5bzcA/edit#heading=h.qex63c29z2to
[2] https://docs.google.com/document/d/1v8BxI07u-mby5f5rCruwF7odSXgb9G8-C9W5hQtSIAg/edit#heading=h.xw1gqgyqs5b

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

File: istio-control/istio-discovery/templates/configmap.yaml [hunk 1]
```diff
@@ -211,5 +211,13 @@ data:
       envoyAccessLogService:
         address: {{ .Values.global.proxy.envoyAccessLogService.host }}:{{ .Values.global.proxy.envoyAccessLogService.port }}
     {{- end}}
+
+    {{- if .Values.pilot.enableCertificateController }}
```
Comment (costinm) [inline istio-control/istio-discovery/templates/configmap.yaml]: Same as above: a 'certificates' section under values.global, no need for an 'enable' API.

File: istio-control/istio-discovery/templates/configmap.yaml [hunk 2]
```diff
@@ -211,5 +211,13 @@ data:
...
+      certificates:
+        - secretName: istio-webhook-galley
```
Comment (costinm) [inline istio-control/istio-discovery/templates/configmap.yaml]: Let's not use random names for the secretName. 
Istio spiffe secrets follow the pattern: istio.SERVICE_ACCOUNT_NAME

For the DNS certs we should follow a similar convention - istiodns or dns prefix ( I would use dns. to make it clear it's not istio specific - unlike the Spifee accounts that are specific to Istio due the SAN format ). And the second part should be based on either the service account or service name. 

While this won't be enforced, it's good to be organized and have a good style.

File: istio-control/istio-discovery/values.yaml [hunk 3]
```diff
@@ -100,6 +100,10 @@ pilot:
   # Indicate if Galley is enabled to send MCP queries
   useMCP: true
 
+  # When true, DNS certificates will be provisioned through Chiron linked into Pilot.
+  # When false, such mechanism is not enabled.
+  enableCertificateController: false
```
Comment (costinm) [inline istio-control/istio-discovery/values.yaml]: Move this to top level global, and rename to 'certificates:' 

In mesh config - add the content of this section using 'toJSON' - like we do for networks.

Comment (costinm) [reply istio-control/istio-discovery/values.yaml]: And we want this enabled by default - so it gets tested. We can turn it off before the release. ( like we did with sniffing)

Comment (costinm) [reply istio-control/istio-discovery/values.yaml]: I mean: enabled by default now, we'll decide to enable it or not before the release. We need to see what happens on K8S clusters that don't support the API.

File: istio-control/istio-discovery/templates/clusterrole.yaml [hunk 4]
```diff
@@ -34,5 +34,16 @@ rules:
 - apiGroups: [""]
   resources: ["endpoints", "pods", "services", "namespaces", "nodes", "secrets"]
   verbs: ["get", "list", "watch"]
+{{ if .Values.pilot.enableCertificateController }}
```
[the remark this answers is not in the document]
Comment (lei-tang) [reply istio-control/istio-discovery/templates/clusterrole.yaml]: Done: addressed in https://github.com/istio/installer/pull/424; please review it.

File: istio-control/istio-discovery/templates/configmap.yaml (hunk 1 above)
Comment (lei-tang) [reply istio-control/istio-discovery/templates/configmap.yaml]: Done: addressed in https://github.com/istio/installer/pull/424; please review it.

File: istio-control/istio-discovery/templates/configmap.yaml (hunk 2 above)
Comment (lei-tang) [reply istio-control/istio-discovery/templates/configmap.yaml]: Done: addressed in https://github.com/istio/installer/pull/424; please review it.

File: istio-control/istio-discovery/values.yaml (hunk 3 above)
Comment (lei-tang) [reply istio-control/istio-discovery/values.yaml]: Done: addressed in https://github.com/istio/installer/pull/424; please review it.

**Pull request merged**